### PR TITLE
fix templates error and adding prettier and lint with github-actions

### PR DESCRIPTION
**Pull request by ghost**
- Added github-actions pipeline
- Added prettier to format code automatically as pre-commit
- Added test to verify if ejs templates are valid
- Resolved bug moving templates folder under generators/app (as yeoman standard)

### CODE REVIEW

**Comment by ghost** [comment]
There is an error on coveralls task. The root case could be that for publishing report you have to set the 
COVERALLS_REPO_TOKEN variable in this repo: https://github.com/lemurheavy/coveralls-public/issues/632 .

So for now I commented that task in github actions pipeline.

File: .github/workflows/greetings.yml [hunk 1]
```diff
@@ -6,8 +6,8 @@ jobs:
   greeting:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/first-interaction@v1
-      with:
-        repo-token: ${{ secrets.GITHUB_TOKEN }}
-        issue-message: 'Thank you for reporting your first issue. My master will take a look at his earliest convenience. - GitHub Bot'
-        pr-message: 'Thank you for submitting your first PR. My master will take a look at his earliest convenience. - GitHub Bot'
+      - uses: actions/first-interaction@v1
```
Comment (vivekmore) [inline .github/workflows/greetings.yml:9]: is this change as a result of the new pre-commit hook?

File: README.md [hunk 2]
```diff
@@ -1,6 +1,9 @@
 # Generate Navigational Components With Ease !!!
-[![NPM version][npm-image]][npm-url] [![Build Status][travis-image]][travis-url] [![Dependency Status][daviddm-image]][daviddm-url] [![Coverage Status][coveralls-image]][coveralls-url]
 
+[![NPM version][npm-image]][npm-url]
```
Comment (vivekmore) [inline README.md:3]: does this display the badges one below the other? or in a horizontal line?
I would like to keep it on a single line.
So if this doesn't do that please revert this change.

File: README.md [hunk 3]
```diff
@@ -62,16 +68,20 @@ npm update -g generator-jhipster-nav-element
 yarn global upgrade generator-jhipster-nav-element
 ```
 
+# Contributors ✨
```
Comment (vivekmore) [inline README.md:71]: not sure I want this section just yet.

File: README.md [hunk 4]
```diff
@@ -62,16 +68,20 @@ npm update -g generator-jhipster-nav-element
...
-Apache-2.0 © [Vivek More]
+# License
 
+Apache-2.0 © [Vivek More](https://github.com/vivekmore)
 
 [npm-image]: https://img.shields.io/npm/v/generator-jhipster-nav-element.svg
 [npm-url]: https://npmjs.org/package/generator-jhipster-nav-element
-[travis-image]: https://travis-ci.org/vivekmore/generator-jhipster-nav-element.svg?branch=master
-[travis-url]: https://travis-ci.org/vivekmore/generator-jhipster-nav-element
+[github-actions-generator-ci-image]: https://github.com/jhipster/jhipster-quarkus/workflows/Generator%20CI/badge.svg
```
Comment (vivekmore) [inline README.md:83]: I don't want to lose the travis build info, please retain that

File: _DEPRECATED.travis.yml [hunk 5]
```diff
@@ -2,15 +2,15 @@ os:
   - linux
 language: node_js
 node_js:
-  - '10.16.1'
+  - "10.16.1"
```
Comment (vivekmore) [inline _DEPRECATED.travis.yml:5]: why deprecated?

File: docs/CODE_OF_CONDUCT.md [hunk 6]
```diff
@@ -8,19 +8,19 @@ In the interest of fostering an open and welcoming environment, we as contributo
 
 Examples of behavior that contributes to creating a positive environment include:
 
-* Using welcoming and inclusive language
-* Being respectful of differing viewpoints and experiences
-* Gracefully accepting constructive criticism
-* Focusing on what is best for the community
-* Showing empathy towards other community members
+- Using welcoming and inclusive language
```
Comment (vivekmore) [inline docs/CODE_OF_CONDUCT.md:11]: why this change? are dashes better than asterisks?

File: docs/PUBLISHING.md [hunk 7]
```diff
@@ -1,63 +1,67 @@
...
     git push origin master
     ```
 
 # Troubleshooting
+
 See [this page](https://docs.npmjs.com/getting-started/working_with_tokens#how-to-create-a-new-full-permission-token) for token generation
-    ```bash
-    npm token create
-    ```
+`bash npm token create`
```
Comment (vivekmore) [inline docs/PUBLISHING.md:67]: revert this please

File: docs/PUBLISHING.md [hunk 8]
```diff
@@ -1,63 +1,67 @@
...
 
-7. Merge the upgrade branch to develop
     ```bash
     git co develop
     git merge upgrade --no-ff
     git push origin develop
     ```
 
-8. Merge the tag to master (e.g. to merge a newly created v1.0.0 tag to master)
+8.  Merge the tag to master (e.g. to merge a newly created v1.0.0 tag to master)
```
Comment (vivekmore) [inline docs/PUBLISHING.md:57]: I don't see the value in changes in this file. Kindly revert.

File: generators/app/about-us/about-us-ng-template-prompter.js [hunk 9]
```diff
@@ -14,19 +14,19 @@ function askQuestions(generator) {
         {
             type: 'input',
             name: 'navElementKey',
-            message: 'What would you like the navigation element id to be? (e.g. aboutUs, about_us)',
+            message:
```
Comment (vivekmore) [inline generators/app/about-us/about-us-ng-template-prompter.js:17]: please revert this.

File: generators/app/about-us/about-us-ng-template-writer.js [hunk 10]
```diff
@@ -16,22 +16,30 @@ function write(generator) {
...
+    const s = generator.navElementKey
+        .trim()
+        .replace(' ', '-')
+        .replace('_', '-');
     generator.translationKeyMenu = _.kebabCase(s).toLowerCase();
     generator.componentStartCase = _.startCase(s);
 
-    const prefix = jhipsterAppConfig.jhiPrefix ? `${_.kebabCase(jhipsterAppConfig.jhiPrefix)}-` : '';
+    const prefix = jhipsterAppConfig.jhiPrefix
+        ? `${_.kebabCase(jhipsterAppConfig.jhiPrefix)}-`
```
Comment (vivekmore) [inline generators/app/about-us/about-us-ng-template-writer.js:27]: please revert any formatting changes (in all files). I don't think I like this format.

**Review by vivekmore — CHANGES_REQUESTED** [review]
I think it'll be best if you submitted a PR for each point
- I personally don't like the new format being applied
- some changes are redundant and add no value (e.g. spaces in *.MD files)
- some sections are not necessary yet in my opinion  (e.g. contributors, contributions)

File: .github/workflows/greetings.yml (hunk 1 above)
Comment (ghost) [reply .github/workflows/greetings.yml:9]: The file is the same of your. You can check, it is only formatted.

File: README.md (hunk 2 above)
Comment (ghost) [reply README.md:3]: The badges keep to be displayed in horizontal line. The README.md does not displayed white line for badges. I have formatted it only for more visibility of the lines

File: README.md (hunk 3 above)
Comment (ghost) [reply README.md:71]: ok, I remove it

File: README.md (hunk 4 above)
Comment (ghost) [reply README.md:83]: ok

File: _DEPRECATED.travis.yml (hunk 5 above)
Comment (ghost) [reply _DEPRECATED.travis.yml:5]: According jhipster standard, now all the build will be on github actions. So I have renamed the travis file to avoid to start automatically build also on travis.

File: docs/CODE_OF_CONDUCT.md (hunk 6 above)
Comment (ghost) [reply docs/CODE_OF_CONDUCT.md:11]: The standard for a bulleted list using mark down is the dash. Asterisk is preferred for the bold words. However, the file is rendered as your previous.

File: docs/PUBLISHING.md (hunk 7 above)
Comment (ghost) [reply docs/PUBLISHING.md:67]: The result rendered is the same. It is been formatted automatically

File: docs/PUBLISHING.md (hunk 8 above)
Comment (ghost) [reply docs/PUBLISHING.md:57]: The formatter has removed an unuseful white space.

File: generators/app/about-us/about-us-ng-template-prompter.js (hunk 9 above)
Comment (ghost) [reply generators/app/about-us/about-us-ng-template-prompter.js:17]: Please view the entire file. Github has underlined only a part, and it is not correct. There is the message, is not blank, the formatter has only add this in a new line because the string is too long.

File: generators/app/about-us/about-us-ng-template-writer.js (hunk 10 above)
Comment (ghost) [reply generators/app/about-us/about-us-ng-template-writer.js:27]: I have understood that you don't like a formatter.. It was useful to visualize better files and code (and it is also used bu the generator-jhipster). But if you don't want, no problem :)

**Comment by ghost** [comment]
I have split in PR #37 and #38

File: .github/workflows/greetings.yml (hunk 1 above)
Comment (vivekmore) [reply .github/workflows/greetings.yml:9]: Yes that's what I was asking about. 
I don't see why it needed to be formatted.
I don't want prettier integration (yet)

File: _DEPRECATED.travis.yml (hunk 5 above)
Comment (vivekmore) [reply _DEPRECATED.travis.yml:5]: This module supports projects built with JHipster.
However, the coding/ci standards are independent of that project.

File: .github/workflows/greetings.yml (hunk 1 above)
Comment (ghost) [reply .github/workflows/greetings.yml:9]: The prettier task automatically apply its spaces and indent rules, according its standard

File: docs/PUBLISHING.md (hunk 7 above)
Comment (vivekmore) [reply docs/PUBLISHING.md:67]: The result showed the word bash in it which was incorrect.
Again - like I mentioned for instances like these, I want to avoid having automatic formatting for now.

File: _DEPRECATED.travis.yml (hunk 5 above)
Comment (ghost) [reply _DEPRECATED.travis.yml:5]: Now, if you generate a new module with the generator-jhipster-module, it is supported only github actions for ci

File: generators/app/about-us/about-us-ng-template-prompter.js (hunk 9 above)
Comment (vivekmore) [reply generators/app/about-us/about-us-ng-template-prompter.js:17]: I know. I did look at the entire file.
I don't think I want this file formatted.
Kindly revert any changes resulting from a format or prettier changes

File: generators/app/about-us/about-us-ng-template-writer.js (hunk 10 above)
Comment (vivekmore) [reply generators/app/about-us/about-us-ng-template-writer.js:27]: Like I said this project supports projects built with JHipster.
However, the coding guidelines and other aspects of this project are independent of JHipster project.

File: docs/PUBLISHING.md (hunk 7 above)
Comment (ghost) [reply docs/PUBLISHING.md:67]: the rendering is the same of the old, the automatic formatter knows that bash is an inuseful in your case. However do not apply for formatter code, I remove it.

**Comment by vivekmore** [comment]
> I have split in PR #37 and #38

Kindly give descriptions to your PRs - it helps the reviewer (me) to identify what changes are made for.

**Comment by ghost** [comment]
Do not apply for formatter changes, I remove it. Loot to the other PR

File: generators/app/about-us/about-us-ng-template-writer.js (hunk 10 above)
Comment (amanganiello90) [reply generators/app/about-us/about-us-ng-template-writer.js]: > 
> 
> > I have split in PR #37 and #38
> 
> Kindly give descriptions to your PRs - it helps the reviewer (me) to identify what changes are made for.

The commit message refers to the issue. It is the solution, so it is already an explanation.

**Comment by ghost** [comment]
> 
> 
> > I have split in PR #37 and #38
> 
> Kindly give descriptions to your PRs - it helps the reviewer (me) to identify what changes are made for.

The commit message refers to the issue. It is the solution, so it is already an explanation.

**Comment by vivekmore** [comment]
> > > I have split in PR #37 and #38
> > 
> > 
> > Kindly give descriptions to your PRs - it helps the reviewer (me) to identify what changes are made for.
> 
> The commit message refers to the issue. It is the solution, so it is already an explanation.

Please do not argue here.
This is open source, if you want to contribute please be respectful and mindful of the original authors processes.

I need PRs to have descriptions for internal tracking and other housekeeping. I don't need to justify.

**Comment by ghost** [comment]
> 
> 
> > > > I have split in PR #37 and #38
> > > 
> > > 
> > > Kindly give descriptions to your PRs - it helps the reviewer (me) to identify what changes are made for.
> > 
> > 
> > The commit message refers to the issue. It is the solution, so it is already an explanation.
> 
> Please do not argue here.
> This is open source, if you want to contribute please be respectful and mindful of the original authors processes.
> 
> I need PRs to have descriptions for internal tracking and other housekeeping. I don't need to justify.

My arguments had no intention of criticism or offense, they were only simple explanations. Otherwise I offer my sincere apologies if I have been misunderstood.

Hovewer, you can decline this PR, I understood that you doesn't want the prettier, so I have split the changes in two PR:
- #37 that fixed the error of the templates folder. But now I have seen that you have just released your hotfix
- #38 that adds features of the github actions. It includes only a commit with the issue number in the message: #30 .. and that is the explanation, the issue title that it refers.

So consider the last. You can close this, it's ok for me :)
